### PR TITLE
Simplify pipeline's profile

### LINKED ISSUE #3530: Simplify pipeline's profile
## Feature request

In pipeline engine, profile's hierarchy (including 5 levels)  is as follows:

Profile
* Summy
* Fragment-1
    * FragmentInstance-1
        * Pipeline-1
            * PipelineDriver-1
                * Operator-1
                * Operator-2
                * Operator-3
            * PipelineDriver-2
            * PipelineDriver-3
        * Pipeline-2
        * Pipeline-3
    * FragmentInstance-2
    * FragmentInstance-3
* Fragment-2
* Fragment-3

Profile's size grows rapidly by increasing degree of paralleliasm. For some TPC-DS querys, the total num of rows reaches amazing 200k lines which is really hard to do further analysis.

Considering the struct of profile in the same level are isomorphic. E.g. All the `FragmentInstance`'s profile within same `Fragment` are isomorphic, All the `PipelineDriver`'s profile within same `Pipeline` are isomorphic. So we can merge these isomorphic profiles into one. 

The hierarchy of merged `Profile` are like as follows, as you can see, `FragmentInstance` and `PipelineDriver` are removed from the hierarchy.

Profile
* Summy
* Fragment-1
    * Pipeline-1
        * Operator-1
        * Operator-2
        * Operator-3      
    * Pipeline-2
    * Pipeline-3
* Fragment-2
* Fragment-3

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #3530

## Details

* change session variable `pipeline_profile_mode` from type `string` to `int`
    * `0`: default value. Merge isomorphic profiles and only remain core metrics
    * `1`: Merge isomorphic profiles and contain all the metrics
    * `2`: Keep the original profiles

### CODE REVIEW

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Comment by satanson** [comment]
can the profile show how many instances belongs to the same Fragment? so as the number of drivers belongs to a pipeline?
the # of instances/drivers are sometimes significant.

File: be/src/exec/pipeline/operator.h [hunk 1]
```diff
@@ -141,6 +141,8 @@ class Operator {
     // Which plan node this operator belongs to
     const int32_t _plan_node_id;
     std::shared_ptr<RuntimeProfile> _runtime_profile;
```
Comment (satanson) [inline be/src/exec/pipeline/operator.h:143]: add comment to explain _common_metrics, _unique_metrics and _runtime_profile

**Comment by satanson** [comment]
run starrocks_fe_unittest

File: be/src/exec/pipeline/operator.h (hunk 1 above)
Comment (liuyehcf) [reply be/src/exec/pipeline/operator.h:143]: ok

**Comment by liuyehcf** [comment]
> can the profile show how many instances belongs to the same Fragment? so as the number of drivers belongs to a pipeline? the # of instances/drivers are sometimes significant.

Ok, I can add another counter named `InstanceNum` attached to `Fragment`'s profile

**Review by satanson — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**